moshi/src/main/java/com/squareup/moshi/BufferedSourceJsonReader.java

```diff
@@ -237,6 +238,13 @@
 
   private int doPeek() throws IOException {
     int peekStack = stack[stackSize - 1];
+
+    if (peekStack == JsonScope.EMPTY_DOCUMENT
+            && source.request(BYTE_ORDER_MARK.size())
+            && buffer.indexOf(BYTE_ORDER_MARK) == 0) {
```

Comment:
This was released in 1.9.0. I'm tempted to say that since it's so easy to test and skip the BOM that we don't really need to add it to Moshi.